osc-rest-server/src/main/java/org/osc/core/broker/rest/server/api/VirtualizationConnectorApis.java

```diff
@@ -492,5 +510,99 @@ public Response updateVirtualSecurityPolicyBindings(@Context HttpHeaders headers
+                                             @ApiParam(value = "The Service Function Chain Id") @PathParam("sfcId") Long sfcId) {
+        logger.info("getting Service Function Chain " + sfcId);
+        this.userContext.setUser(OscAuthFilter.getUsername(headers));
+        GetDtoFromEntityRequest getDtoRequest = new GetDtoFromEntityRequest();
+        getDtoRequest.setEntityId(sfcId);
+        getDtoRequest.setEntityName("ServiceFunctionChain");
+        GetDtoFromEntityServiceApi<ServiceFunctionChainDto> getDtoService = this.getDtoFromEntityServiceFactory.getService(ServiceFunctionChainDto.class);
+        ServiceFunctionChainDto dto = this.apiUtil.submitBaseRequestToService(getDtoService, getDtoRequest).getDto();
 
+        return dto;
+    }
+    
+    @ApiOperation(value = "Deletes a Service Function Chain",
+            notes = "Delete a Service Function Chain owned by a Virutlaization Connector")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successful operation"),
+            @ApiResponse(code = 400, message = "In case of any error", response = ErrorCodeDto.class) })
+    @Path("/{vcId}/serviceFunctionChain/{sfcId}")
+    @DELETE
+    public Response deleteServiceFunctionChain(@Context HttpHeaders headers, 
+            @ApiParam(value = "The Virtualization Connector Id") @PathParam("vcId") Long vcId,
+            @ApiParam(value = "The Service Function Chain Id") @PathParam("sfcId") Long sfcId) {
+        logger.info("Deleting Service Function Chain  " + sfcId);
+        this.userContext.setUser(OscAuthFilter.getUsername(headers));
+        return this.apiUtil.getResponseForBaseRequest(this.deleteServiceFunctionChainService, new BaseIdRequest(sfcId, vcId));
+    }
+    
+    @ApiOperation(value = "Update a Service Function Chain",
+            notes = "Update a Service Fucntion Chain owned by Virtualization Connector provided and kicks off a " + "sync job",
+            response = BaseJobResponse.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Successful operation"),
+            @ApiResponse(code = 400, message = "In case of any error", response = ErrorCodeDto.class) })
+    @Path("/{vcId}/serviceFunctionChain/{sfcId}")
+    @PUT
+    public Response updateServiceFunctionChain(@Context HttpHeaders headers,
+                                        @ApiParam(value = "The Virtualization Connector Id") @PathParam("vcId") Long vcId,
+                                        @ApiParam(value = "The Service Function Chain Id") @PathParam("sfcId") Long sfcId,
+                                        @ApiParam(value = "List of SFC vsIds", required = false) List<Long> vsIds) {
+        logger.info("Update Service Function Chain ...");
+        this.userContext.setUser(OscAuthFilter.getUsername(headers));
+        ServiceFunctionChainDto sfcDto = new ServiceFunctionChainDto();
+        this.apiUtil.setIdAndParentIdOrThrow(sfcDto, sfcId, vcId, "Service Function Chain");
```

Comment:
we need to get the chain dto from the user and then this method validates the values